tensorflow/core/platform/ctstring_internal.h

```diff
@@ -162,6 +162,14 @@ static inline size_t TF_TString_ToInternalSizeT(size_t size,
 #else   // TF_TSTRING_LITTLE_ENDIAN
   // 0xFF000000 or 0xFF00000000000000 depending on platform
   static const size_t mask = ~((~(size_t)0) >> 8);
+#if defined(__s390x__)
```

Comment:
Judging from the description, this issue is not only present for s390x, we have the same thing for any big-endian system with 64-bit `size_t`.  In the case of big-endian and `TF_TSTR_OFFSET`, instead of putting the type in the MSB byte, we should *always* be putting it in the 4th byte from the right.